x-pack/plugins/infra/public/pages/metrics/hosts/components/controls_content.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { ControlGroupInput } from '@kbn/controls-plugin/public';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { Filter, TimeRange } from '@kbn/es-query';
+import { LazyControlsRenderer } from './lazy_controls_renderer';
+import { useControlPanels } from '../hooks/use_control_panels_url_state';
+
+interface Props {
+  timeRange: TimeRange;
+  dataViewId: string;
+  filters: Filter[];
+  query: {
+    language: string;
+    query: string;
+  };
+  setPanelFilters: React.Dispatch<React.SetStateAction<null | Filter[]>>;
+}
+
+export const ControlsContent: React.FC<Props> = ({
+  timeRange,
+  dataViewId,
+  query,
+  filters,
+  setPanelFilters,
+}) => {
+  const { setControlPanels, controlPanel } = useControlPanels(dataViewId);
+
+  return (
+    <LazyControlsRenderer
+      getCreationOptions={async ({ addDataControlFromField }) => {
```

Comment:
**nit**: since this function does not wait for any async operation, you can safely remove the `async` and keep your function synchronous.
Also, you don't need to restructure the args if not necessary, and to improve the readability of the return statement you could abstract the object into a factory function out of the return statement, something like this:
```ts
const makeCreationOption = ({dataViewId, timeRange, filters, query}): Partial<ControlGroupInput> => ({
  id: dataViewId,
  timeRange,
  viewMode: ViewMode.VIEW,
  filters,
  query,
  chainingSystem: 'HIERARCHICAL',
  controlStyle: 'oneLine',
  defaultControlWidth: 'small',
  panels: controlPanel,
});


...

const handleOptionCreation = () => makeCreationOption({dataViewId, timeRange, filters, query})

return (
  <LazyControlsRenderer 
    getCreationOptions={handleOptionCreation}
    onEmbeddableLoad={...}
...
```

Reply:
The function (`getCreationOptions`) is defined to return a promise, that's why it has `async`.  The methods that we need to `wait` are not used in this case but the returned type is still a promise.